arch/x86/core/ia32/intstub.S

```diff
@@ -180,6 +180,21 @@ alreadyOnIntStack:
 	popl	%eax
 #endif
 
+#if defined(CONFIG_X86_RUNTIME_IRQ_STATS)
+	/*
+	 *  The runtime_irq_stats() function is optional, marked as a
+	 *  weak symbol here, it will get valued zero if not defined,
+	 *  platform with this enabled config will implement this.
+	 */
+	.weak	runtime_irq_stats
+	cmpl	$0, runtime_irq_stats
+	je	skip_runtime_irq_stats
```

Comment:
There is no need to test for existence of this function. Turning on the kconfig implies that this function already exists. So there is no need to waste time on testing.

Reply:
Updated, thanks